### PR TITLE
Replace Newtonsoft.Json by System.Text.Json

### PR DESCRIPTION
**Pull request by kinosang**
Addressed #69

### CODE REVIEW

File: src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs [hunk 1]
```diff
@@ -96,12 +97,14 @@ public class SteamAuthenticationHandler : OpenIdAuthenticationHandler<SteamAuthe
                 throw new HttpRequestException("An error occurred while retrieving the user profile from Steam.");
             }
 
-            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+            var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
```
Comment (martincostello) [inline src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs]:
```suggestion
            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
```

File: src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs [hunk 2]
```diff
@@ -96,12 +97,14 @@ public class SteamAuthenticationHandler : OpenIdAuthenticationHandler<SteamAuthe
...
 
             // Try to extract the profile name of the authenticated user.
-            var profile = payload.Value<JObject>(SteamAuthenticationConstants.Parameters.Response)
-                                ?.Value<JArray>(SteamAuthenticationConstants.Parameters.Players)
-                            ?[0]?.Value<string>(SteamAuthenticationConstants.Parameters.Name);
+            var profile = payload.RootElement.GetProperty(SteamAuthenticationConstants.Parameters.Response)
+                                .GetProperty(SteamAuthenticationConstants.Parameters.Players)
+                                .EnumerateArray()
+                                .First()
```
Comment (martincostello) [inline src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs]: Should we harden this against the array being empty?

Comment (kinosang) [reply src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs]: @martincostello `JsonElement` is not nullable, if use `FirstOrDefault`, it cannot be access conditionally.

Comment (martincostello) [reply src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs]: But `First()` will throw if there are none?

**Review by martincostello — APPROVED** [review]

**Comment by martincostello** [comment]
Thanks @kinosang!

**Pull request merged**